src/java/org/jivesoftware/openfire/muc/spi/LocalMUCRoomManager.java

```diff
@@ -0,0 +1,49 @@
+package org.jivesoftware.openfire.muc.spi;
+
+import java.util.Collection;
+import java.util.Date;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.jivesoftware.openfire.event.GroupEventDispatcher;
+import org.jivesoftware.openfire.muc.MUCRoom;
+
+/**
+ * this class supports the simple LocalMUCRoom management including remove,add and query.
+ * @author wuchang
+ * @email 583424568@qq.com
+ * 2016-1-14
+ */
+public class LocalMUCRoomManager {
+    private Map<String, LocalMUCRoom> rooms = new ConcurrentHashMap<>();
+     
+    public int getNumberChatRooms(){
+    	return rooms.size();
+    }
+    public void addRoom(String roomname,LocalMUCRoom room){
+    	rooms.put(roomname, room);
```

Comment:
In #518 I mentioned a case when the room is added in GroupEventDispatcher and is not removed.

That's why the GroupEventDispatcher is called inside the LocalMUCRoom constructor. 

So, I think the GroupEventDispatcher.addListener(room) should be called here, not in https://github.com/VicoWu/Openfire/blob/master/src/java/org/jivesoftware/openfire/muc/spi/LocalMUCRoom.java#L385